packages/core-data/src/private-selectors.ts

```diff
@@ -92,7 +92,7 @@ export function getEntityRecordPermissions(
 	name: string,
 	id: string
 ) {
-	return getEntityRecordsPermissions( state, kind, name, id )[ 0 ];
+	return getEntityRecordsPermissions( state, kind, name, [ id ] )[ 0 ];
```

Comment:
We need to update the `getEntityRecordsPermissions` types instead of changing the passed argument. It accepts `string|string[]`.

I know it's a bit odd, but that's due to the limitations of memoized selectors. Current change regresses #64091.

## Screenshot

![CleanShot 2024-12-09 at 21 37 59](https://github.com/user-attachments/assets/39c10f92-0c28-4b72-bf1b-42c6e01af2f7)

Reply:
I had done exactly that here https://github.com/WordPress/gutenberg/pull/67667#discussion_r1873359180

Reply:
But it was reverted in 9a5a859611939df9df7144b2e06e62de921d1517, causing the regression.

Also, I'm not sure if I fully understand the reason for [reverting](https://github.com/WordPress/gutenberg/pull/67667#discussion_r1873359180). These are private selectors; they can only be consumed in the core.

Reply:
I'll create a follow up PR to address that. Thanks.

Reply:
Thank you, @manzoorwanijk!

Reply:
Here is the PR 

- #67770